tt_metal/common/work_split.cpp

```diff
@@ -148,6 +148,114 @@ CoreRangeSet num_cores_to_corerangeset(
     return num_cores_to_corerangeset({0, 0}, target_num_cores, grid_size, row_wise);
 }
 
+CoreRangeSet num_cores_to_corerangeset_in_subcoregrids(
+    const CoreCoord start_core,
+    const uint32_t target_num_cores,
+    const CoreRangeSet subcoregrids,
+    const bool row_wise = false) {
+    // If target_num_cores is 0 or input_corerangeset is empty, return empty CoreRangeSet
+    TT_FATAL(target_num_cores > 0, "Target number of cores must be greater than 0");
+    TT_FATAL(
+        target_num_cores <= subcoregrids.num_cores(),
+        "Target number of cores {} is greater than total number of available cores {}",
+        target_num_cores,
+        subcoregrids.num_cores());
+
+    // Validate that the start core is contained within the entire CoreRangeSet
+    TT_FATAL(subcoregrids.contains(start_core), "Start core must be contained within the input CoreRangeSet");
+
+    std::vector<CoreRange> result_coreranges;
+    bool start_core_found = false;
+    CoreCoord current_start_core = start_core;
+    CoreCoord current_end_core = start_core;
+    uint32_t remaining_cores = target_num_cores;
+
+    for (const auto& subcoregrid : subcoregrids.ranges()) {
+        // Check if start_core is in the current subcoregrid
+        if (subcoregrid.contains(start_core)) {
+            start_core_found = true;
+        } else {
+            if (!start_core_found) {
+                continue;
+            } else {
+                current_start_core = subcoregrid.start_coord;
+                current_end_core = current_start_core;
+            }
+        }
+
+        uint32_t subcoregrid_width = subcoregrid.grid_size().x;
+        uint32_t subcoregrid_height = subcoregrid.grid_size().y;
+
+        if (row_wise) {
+            for (uint32_t y = current_start_core.y; y <= subcoregrid.end_coord.y; ++y) {
```

Comment:
could you split this function into two subfunctions? row_wise and col_wise. this for loop is very large.
YOu simple can create two lambdas at least and call on of them in the cycle.

Reply:
@dmakoviichuk-tt converted them to lambdas now.